docs/building-your-application/routing/middleware.md

```diff
@@ -117,6 +117,42 @@ export default async function middleware(request) {
 
 However, this is not the best way to serve assets. You can put the static files directly inside the `public` folder. More information [here](/building-your-application/routing/static-assets).
 
+## Redirect
+
+You can redirect the user to another page by returning a `Response` object with a `302` status code and a `Location` header.
+
+```ts
+export default async function middleware(request) {
+  if (request.url.pathname === "/old") {
+    return new Response("", {
+      status: 302,
+      headers: {
+        Location: "/new",
+      },
+    });
+  }
+}
+```
+
+When the user visits `/old`, they will be redirected to `/new`.
+
+## Rewrite
+
+You can rewrite the request by modifying the `finalURL` of the `Request` object.
+
+```ts
+export default async function middleware(request) {
+  if (request.url.pathname === "/old") {
+    // Rewrite: /old -> /new
+    request.finalURL = new URL("/new", request.finalURL).toString();
+    // Continue processing the request
+    return;
```

Comment:
Do we really need this return? Maybe there is some logic after this.

Reply:
In this example it is not necessary because there is nothing else after that conditon. What is necessary is that it does not return anything to continue processing. Maybe can be improved just with a comment:

```ts
export default async function middleware(request) {
  if (request.url.pathname === "/old") {
    // Rewrite: /old -> /new
    request.finalURL = new URL("/new", request.finalURL).toString();
    // Continue processing the request
    return;
  }

  // ... Manage other routes   <--- This comment
}
```

Reply:
Sounds better yeah. Also, can we add an explanation of what will happen if you do:

```typescript
req.finalURL = new URL('/admin', req.finalURL).toString();
    return new Response('', {
      status: 302,
      headers: {
        Location: new URL('/admin', req.url).toString(),
      },
    });
```

Reply:
In this case is returning the response and ignoring the rewrite. I'm going to add it